staging/src/k8s.io/apiserver/pkg/cel/mutation/optional_test.go

```diff
@@ -0,0 +1,134 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mutation
+
+import (
+	"reflect"
+	"strings"
+	"testing"
+)
+
+// TestCELOptional is an exploration test to demonstrate how CEL optional library
+// behave for the use cases that the mutation library requires.
+func TestCELOptional(t *testing.T) {
+	for _, tc := range []struct {
+		name                 string
+		expression           string
+		expectedValue        any
```

Comment:
Could you make this a ref.Val and use the `.Equal(other)` method to determine equality with the expected output. The `nil` comparisons sort of hide that the intent is supposed to be `Optional.None` in some of these tests.

Reply:
Good point! I've converted all of the tests to use `ref.Val` and `Equal` for assertions. Thank you.